Kyle_Marienthal/DJANGO/ninjagold/apps/ninja/views.py

```diff
@@ -0,0 +1,49 @@
+from django.shortcuts import render, redirect, HttpResponse
+import random
+
+# Create your views here.
+def index(request):
+    #instantiate default value for request.session['activities']
+    if 'activities' not in request.session:
+        request.session['activities'] = []
+
+    # Instantiate default value for request.session['gold']
+    if 'gold' not in request.session:
+        request.session['gold'] = 0
+    context = {
+        'activities': request.session['activities'],
+        'gold': request.session['gold'],
+    }
+    request.session.modified = True
+    return render(request, "ninja/index.html", context)
+
+def process_money(request):
```

Comment:
In this method you're overwritting session activities on each request. It may be better to append onto the activities instead so you could constantly track all activities. request.session['activities'].append(string_message)